scripts/Engineering/gui/engineering_diffraction/settings/test/test_settings_helper.py

```diff
@@ -0,0 +1,112 @@
+        settings = QSettings()
+        settings.clear()
+        rmtree(dirname(settings.fileName()))
+
+    def setUp(self):
+        settings = QSettings()
+        settings.clear()
+
+    @classmethod
+    def setUpClass(cls):
+        QCoreApplication.setApplicationName("test1")
+        QCoreApplication.setOrganizationName("org1")
+        QSettings.setDefaultFormat(QSettings.IniFormat)
+
+    def test_set_setting_with_string(self):
+        set_setting(GROUP, PREFIX, "something", "value")
+
+        settings = QSettings()
+        settings.beginGroup(GROUP)
+        returned = settings.value(PREFIX + "something")
+        settings.endGroup()
+        self.assertEqual(returned, "value")
+
+    def test_set_setting_with_bool_false(self):
+        set_setting(GROUP, PREFIX, "something", False)
+
+        settings = QSettings()
+        settings.beginGroup("CustomInterfaces")
+        returned = settings.value("EngineeringDiffraction2/" + "something", type=bool)
+        settings.endGroup()
+        self.assertEqual(returned, False)
+
+    def test_set_setting_with_bool_true(self):
+        set_setting(GROUP, PREFIX, "something", True)
+
+        settings = QSettings()
+        settings.beginGroup("CustomInterfaces")
+        returned = settings.value("EngineeringDiffraction2/" + "something", type=bool)
+        settings.endGroup()
+        self.assertEqual(returned, True)
+
+    def test_set_setting_with_int(self):
+        set_setting(GROUP, PREFIX, "something", 10)
+
+        settings = QSettings()
+        settings.beginGroup(GROUP)
+        returned = settings.value(PREFIX + "something", type=int)
+        settings.endGroup()
+        self.assertEqual(returned, 10)
+
+    def test_get_setting_with_string(self):
+        settings = QSettings()
+        settings.beginGroup(GROUP)
+        settings.setValue(PREFIX + "something", "value")
+        settings.endGroup()
+
+        self.assertEqual(get_setting(GROUP, PREFIX, "something"), "value")
+
+    def test_get_setting_with_bool_false(self):
+        settings = QSettings()
+        settings.beginGroup(GROUP)
+        settings.setValue(PREFIX + "something", False)
+        settings.endGroup()
+
+        self.assertEqual(get_setting(GROUP, PREFIX, "something", return_type=bool), False)
+
+    def test_get_setting_with_bool_true(self):
+        settings = QSettings()
+        settings.beginGroup(GROUP)
+        settings.setValue(PREFIX + "something", True)
+        settings.endGroup()
+
+        self.assertEqual(get_setting(GROUP, PREFIX, "something", return_type=bool), True)
+
+    def test_get_setting_with_int(self):
+        settings = QSettings()
+        settings.beginGroup(GROUP)
+        settings.setValue(PREFIX + "something", 10)
+        settings.endGroup()
+
+        self.assertEqual(get_setting(GROUP, PREFIX, "something", return_type=int), 10)
+
+    def test_get_setting_with_invalid(self):
+        self.assertEqual(get_setting(GROUP, PREFIX, "something"), "")
+
```

Comment:
What happens if you try to get a setting with a return type that cannot be converted from the value? Like getting an `int` from `'text'`. Can we get a test for that?